src/JuMP.jl

```diff
@@ -565,6 +565,22 @@ end
 # GenericRangeConstraint, LinearConstraint
 include("affexpr.jl")
 
+struct VectorAffineConstraint{S <: MOI.AbstractVectorSet} <: AbstractConstraint
+    func::MOI.VectorAffineFunction{Float64}
```

Comment:
Having MOI references here doesn't make too much sense to me. This is a JuMP-level constraint which should have JuMP variables in it. ``Vector{AffExpr}`` could work. It should be consistent with [`LinearConstraint`](https://github.com/JuliaOpt/JuMP.jl/blob/eea38231faaf8ace1e1914d8235759fedc6f336b/src/affexpr.jl#L186). (Maybe `LinearConstraint` is a confusing name since we allow arbitrary sets.)